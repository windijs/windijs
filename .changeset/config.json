{
  "$schema": "https://unpkg.com/@changesets/config@2.1.1/schema.json",
  "changelog": "@changesets/cli/changelog",
  "commit": false,
  "fixed": [["@windijs/core", "windijs"]],
  "linked": [],
  "access": "public",
  "baseBranch": "main",
  "updateInternalDependencies": "patch",
  "ignore": []
}
