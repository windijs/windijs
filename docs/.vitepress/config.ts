import { defineConfig } from "vitepress";
import { mdRenderFilename } from "./markdown";

export default defineConfig({
  lang: "en-US",
  title: "Windi JS",
  cleanUrls: "without-subfolders",
  description: "The Utility-first CSS-in-JS Framework Powered by ES6 Proxy API and TypeScript.",
  markdown: {
    config(md) {
      md.use(mdRenderFilename);
    },
  },
  head: [
    ['link', { rel: "shortcut icon", type: "image/x-icon", href: "/windijs/favicon.ico" }]
  ],
  base: "/windijs/",
  themeConfig: {
    logo: "/logo.png",
    nav: [
      { text: "Guide", link: "/guide/what-is-windijs" },
      { text: "API", link: "/api" },
      { text: "Repl", link: "/repl" },
      { text: "Examples", link: "https://github.com/windijs/windijs/tree/main/examples" },
    ],
    sidebar: [
      {
        text: "Guide",
        collapsible: false,
        items: [
          {
            text: "What is Windi JS?",
            link: "/guide/what-is-windijs",
          },
          {
            text: "Getting Started",
            link: "/guide/getting-started",
          },
          {
            text: "Configuration",
            link: "/guide/configuration",
          },
          {
            text: "Optimization",
            link: "/guide/optimization",
          },
        ],
      },
      {
        text: "Customization",
        collapsible: false,
        items: [
          {
            text: "Theme",
            link: "/customization/theme",
          },
          {
            text: "Utility",
            link: "/customization/utility",
          },
          {
            text: "Variant",
            link: "/customization/variant",
          },
          {
            text: "Handler",
            link: "/customization/handler",
          },
          {
            text: "Loader",
            link: "/customization/loader",
          },
          {
            text: "Namer",
            link: "/customization/namer",
          },
          {
            text: "Plugins",
            link: "/customization/plugins",
          },
        ],
      },
      {
        text: "Integrations",
        collapsible: false,
        items: [
          {
            text: "API",
            link: "/integrations/api",
          },
          // {
          //   text: "CLI",
          //   link: "/integrations/cli",
          // },
          {
            text: "Vite",
            link: "/integrations/vite",
          },
          // {
          //   text: "Rollup",
          //   link: "/integrations/rollup",
          // },
          // {
          //   text: "Webpack",
          //   link: "/integrations/webpack",
          // },
        ],
      },
      {
        text: "Posts",
        collapsible: false,
        items: [
          {
            text: "How Windi JS Works",
            link: "/posts/how-windijs-works",
          },
          {
            text: "Ways of Using Windi JS",
            link: "/posts/ways-of-using-windijs",
          },
        ],
      },
    ],
    editLink: {
      pattern: "https://github.com/windijs/windijs/edit/main/docs/:path",
      text: "Edit this page on GitHub",
    },
    socialLinks: [
      { icon: "discord", link: "https://discord.gg/nJux6Gb52m" },
      { icon: "github", link: "https://github.com/windijs/windijs" },
    ],
    footer: {
      message: "Released under the MIT License.",
      copyright: "Copyright Â© 2022-present Raven Satir",
    },
  },
});
