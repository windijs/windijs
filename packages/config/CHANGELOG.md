# @windijs/config

## 1.0.6

### Patch Changes

- [`134427d6`](https://github.com/windijs/windijs/commit/134427d666a61704de77d52f0e6e4ffbfe085cd3) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: export package.json

* [`99ecf595`](https://github.com/windijs/windijs/commit/99ecf5956d7fa1124c1561d8f58b7a0a2f1af0f2) Thanks [@voorjaar](https://github.com/voorjaar)! - build: switch to es2018 for better tree shaking

* Updated dependencies [[`134427d6`](https://github.com/windijs/windijs/commit/134427d666a61704de77d52f0e6e4ffbfe085cd3), [`99ecf595`](https://github.com/windijs/windijs/commit/99ecf5956d7fa1124c1561d8f58b7a0a2f1af0f2)]:
  - @windijs/shared@1.0.5

## 1.0.5

### Patch Changes

- [`b38b5607`](https://github.com/windijs/windijs/commit/b38b5607dccfb2d5b22af15f1b62af9ad9caf3ae) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: complete config define

* [`3bfacd6e`](https://github.com/windijs/windijs/commit/3bfacd6e745c01f650f1f5e7b65574585ee8c5bc) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: important config with string

* Updated dependencies [[`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75), [`831cae71`](https://github.com/windijs/windijs/commit/831cae71465e13a4d00188cca914710c54cd29cf)]:
  - @windijs/shared@1.0.4

## 1.0.5-beta.3

### Patch Changes

- [`3bfacd6e`](https://github.com/windijs/windijs/commit/3bfacd6e745c01f650f1f5e7b65574585ee8c5bc) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: important config with string

## 1.0.5-beta.2

### Patch Changes

- [`b38b5607`](https://github.com/windijs/windijs/commit/b38b5607dccfb2d5b22af15f1b62af9ad9caf3ae) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: complete config define

## 1.0.5-beta.1

### Patch Changes

- Updated dependencies [[`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75)]:
  - @windijs/shared@1.0.4-beta.1

## 1.0.5-beta.0

### Patch Changes

- Updated dependencies [[`831cae71`](https://github.com/windijs/windijs/commit/831cae71465e13a4d00188cca914710c54cd29cf)]:
  - @windijs/shared@1.0.4-beta.0

## 1.0.4

### Patch Changes

- af3c0c04: feat: add defineConfig interface
- Updated dependencies [939f0398]
  - @windijs/shared@1.0.3

## 1.0.3

### Patch Changes

- fix: should publish CHANGELOG.md
- Updated dependencies
  - @windijs/shared@1.0.2

## 1.0.2

### Patch Changes

- Updated dependencies
  - @windijs/shared@1.0.1

## 1.0.1

### Patch Changes

- 98ef7277: fix: package exports
