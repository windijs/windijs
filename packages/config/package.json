{
  "name": "@windijs/config",
  "version": "1.0.9",
  "description": "@windijs/config",
  "main": "dist/config.js",
  "module": "dist/config.mjs",
  "types": "dist/config.d.ts",
  "files": [
    "dist",
    "CHANGELOG.md"
  ],
  "keywords": [
    "css",
    "css-in-js",
    "windi",
    "windijs",
    "windicss"
  ],
  "exports": {
    "./package.json": "./package.json",
    ".": {
      "import": "./dist/config.mjs",
      "require": "./dist/config.js"
    }
  },
  "author": "Raven Satir",
  "license": "MIT",
  "typedoc": {
    "entryPoint": "./src/index.ts",
    "displayName": "@windijs/config"
  }
}
