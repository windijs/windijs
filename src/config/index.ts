export * from "./colors";
export * from "./bg";
export * from "./flex";
export * from "./grid";
export * from "./border";
export * from "./filter";
export * from "./opacity";
export * from "./sizing";
export * from "./spacing";
export * from "./positioning";
export * from "./interactivity";
export * from "./typography";
