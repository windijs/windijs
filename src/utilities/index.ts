export * from "./base";
export * from "./builder";
export * from "./style";
export * from "./api";
export * from "./bg";
export * from "./font";
export * from "./spacing";
