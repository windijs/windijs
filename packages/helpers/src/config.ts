import { Handler, StyleLoader, StyleObject } from "./types";

import { Theme } from "@windijs/config";

export type DarkModeConfig = "class" | "media" | false;

export type UtilitiesConfig = { [key: string]: Handler<unknown> | Array<Handler<unknown>> | StyleObject | UtilitiesConfig };

export type VariantsConfig = { [key: string]: string | string[] };

export interface Config<T extends object = Theme> {
  // presets?: Config[];
  // prefixer?: boolean;
  // attributify?: boolean | {
  //   prefix?: string
  //   separator?: string
  //   disable?: string[]
  // }
  separator?: string;
  important?: boolean | string;
  darkMode?: DarkModeConfig;
  theme?: T;
  // variantOrder?: string[];
  // plugins?: Plugin[];
  // handlers?: Handlers;
  // corePlugins?: (keyof BaseTheme)[] | string[] | { [ T in keyof BaseTheme ] : boolean } | { [ key:string ] : boolean };
  // prefix?: string;
  // exclude?: RegExp[];
  // alias?: { [key:string]: string };
  // shortcuts?: { [key:string]: Shortcut };
  // purge?: unknown;
  utilities?: UtilitiesConfig;
  variants?: VariantsConfig;
  styleLoader?: StyleLoader;
  [key:string]: any;
}

export function defineConfig (config: Config) {
  return config;
}
