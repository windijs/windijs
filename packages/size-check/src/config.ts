import { paddingConfig } from "@windijs/config";

export { paddingConfig };
