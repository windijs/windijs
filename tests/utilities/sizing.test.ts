import { configHandler, createUtility } from "utilities";
import { heightConfig, widthConfig } from "config";

test("Width", () => {
  const w = createUtility("w")
    .use(configHandler(widthConfig, "width"))
    .init();

  expect(w["1/5"].css).toMatchSnapshot();
  expect(w["3/12"].css).toMatchSnapshot();
  expect(w[10].css).toMatchSnapshot();
  expect(w.md.css).toMatchSnapshot();
  expect(w.screen.css).toMatchSnapshot();
  expect(w.screen.lg.css).toMatchSnapshot();
  expect(w.min.css).toMatchSnapshot();
  expect(w[3.5].css).toMatchSnapshot();
  expect(w.full.css).toMatchSnapshot();
});

test("Min Width", () => {
  const w = createUtility("w")
    .case("min", configHandler(widthConfig, "minWidth"))
    .use(configHandler(widthConfig, "width"))
    .init();

  // TODO: fix this conflict
  expect(w.min.css).toMatchSnapshot();
  expect(w.min["1/5"].css).toMatchSnapshot();
  expect(w.min["3/12"].css).toMatchSnapshot();
  expect(w.min[10].css).toMatchSnapshot();
  expect(w.min.md.css).toMatchSnapshot();
  expect(w.min.screen.css).toMatchSnapshot();
  expect(w.min.screen.lg.css).toMatchSnapshot();
  expect(w.min.min.css).toMatchSnapshot();
  expect(w.min[3.5].css).toMatchSnapshot();
  expect(w.min.full.css).toMatchSnapshot();
});

test("Max Width", () => {
  const w = createUtility("w")
    .case("max", configHandler(widthConfig, "maxWidth"))
    .use(configHandler(widthConfig, "width"))
    .init();

  expect(w.max["1/5"].css).toMatchSnapshot();
  expect(w.max["3/12"].css).toMatchSnapshot();
  expect(w.max[10].css).toMatchSnapshot();
  expect(w.max.md.css).toMatchSnapshot();
  expect(w.max.screen.css).toMatchSnapshot();
  expect(w.max.screen.lg.css).toMatchSnapshot();
  expect(w.max.min.css).toMatchSnapshot();
  expect(w.max[3.5].css).toMatchSnapshot();
  expect(w.max.full.css).toMatchSnapshot();
});

test("Height", () => {
  const h = createUtility("h")
    .use(configHandler(heightConfig, "height"))
    .init();

  expect(h["1/5"].css).toMatchSnapshot();
  expect(h["3/12"].css).toMatchSnapshot();
  expect(h[10].css).toMatchSnapshot();
  expect(h.md.css).toMatchSnapshot();
  expect(h.screen.css).toMatchSnapshot();
  expect(h.screen.lg.css).toMatchSnapshot();
  expect(h.min.css).toMatchSnapshot();
  expect(h[3.5].css).toMatchSnapshot();
  expect(h.full.css).toMatchSnapshot();
});

test("Min Height", () => {
  const h = createUtility("h")
    .case("min", configHandler(heightConfig, "minHeight"))
    .use(configHandler(heightConfig, "height"))
    .init();

  expect(h.min.none.css).toMatchSnapshot();
  expect(h.min["1/5"].css).toMatchSnapshot();
  expect(h.min["3/12"].css).toMatchSnapshot();
  expect(h.min[10].css).toMatchSnapshot();
  expect(h.min.md.css).toMatchSnapshot();
  expect(h.min.screen.css).toMatchSnapshot();
  expect(h.min.screen.lg.css).toMatchSnapshot();
  expect(h.min.min.css).toMatchSnapshot();
  expect(h.min[3.5].css).toMatchSnapshot();
  expect(h.min.full.css).toMatchSnapshot();
});

test("Max Height", () => {
  const h = createUtility("h")
    .case("max", configHandler(heightConfig, "maxHeight"))
    .use(configHandler(heightConfig, "height"))
    .init();

  expect(h.max.none.css).toMatchSnapshot();
  expect(h.max["1/5"].css).toMatchSnapshot();
  expect(h.max["3/12"].css).toMatchSnapshot();
  expect(h.max[10].css).toMatchSnapshot();
  expect(h.max.md.css).toMatchSnapshot();
  expect(h.max.screen.css).toMatchSnapshot();
  expect(h.max.screen.lg.css).toMatchSnapshot();
  expect(h.max.min.css).toMatchSnapshot();
  expect(h.max[3.5].css).toMatchSnapshot();
  expect(h.max.full.css).toMatchSnapshot();
});
