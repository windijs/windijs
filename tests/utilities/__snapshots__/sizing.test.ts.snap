// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Height 1`] = `
Object {
  "height": "20%",
}
`;

exports[`Height 2`] = `
Object {
  "height": "25%",
}
`;

exports[`Height 3`] = `
Object {
  "height": "2.5rem",
}
`;

exports[`Height 4`] = `
Object {
  "height": "28rem",
}
`;

exports[`Height 5`] = `
Object {
  "height": "100vh",
}
`;

exports[`Height 6`] = `
Object {
  "height": "1024px",
}
`;

exports[`Height 7`] = `
Object {
  "height": "min-content",
}
`;

exports[`Height 8`] = `
Object {
  "height": "0.875rem",
}
`;

exports[`Height 9`] = `
Object {
  "height": "100%",
}
`;

exports[`Max Height 1`] = `
Object {
  "maxHeight": "none",
}
`;

exports[`Max Height 2`] = `
Object {
  "maxHeight": "20%",
}
`;

exports[`Max Height 3`] = `
Object {
  "maxHeight": "25%",
}
`;

exports[`Max Height 4`] = `
Object {
  "maxHeight": "2.5rem",
}
`;

exports[`Max Height 5`] = `
Object {
  "maxHeight": "28rem",
}
`;

exports[`Max Height 6`] = `
Object {
  "maxHeight": "100vh",
}
`;

exports[`Max Height 7`] = `
Object {
  "maxHeight": "1024px",
}
`;

exports[`Max Height 8`] = `
Object {
  "maxHeight": "min-content",
}
`;

exports[`Max Height 9`] = `
Object {
  "maxHeight": "0.875rem",
}
`;

exports[`Max Height 10`] = `
Object {
  "maxHeight": "100%",
}
`;

exports[`Max Width 1`] = `
Object {
  "maxWidth": "20%",
}
`;

exports[`Max Width 2`] = `
Object {
  "maxWidth": "25%",
}
`;

exports[`Max Width 3`] = `
Object {
  "maxWidth": "2.5rem",
}
`;

exports[`Max Width 4`] = `
Object {
  "maxWidth": "28rem",
}
`;

exports[`Max Width 5`] = `
Object {
  "maxWidth": "100vw",
}
`;

exports[`Max Width 6`] = `
Object {
  "maxWidth": "1024px",
}
`;

exports[`Max Width 7`] = `
Object {
  "maxWidth": "min-content",
}
`;

exports[`Max Width 8`] = `
Object {
  "maxWidth": "0.875rem",
}
`;

exports[`Max Width 9`] = `
Object {
  "maxWidth": "100%",
}
`;

exports[`Min Height 1`] = `
Object {
  "minHeight": "none",
}
`;

exports[`Min Height 2`] = `
Object {
  "minHeight": "20%",
}
`;

exports[`Min Height 3`] = `
Object {
  "minHeight": "25%",
}
`;

exports[`Min Height 4`] = `
Object {
  "minHeight": "2.5rem",
}
`;

exports[`Min Height 5`] = `
Object {
  "minHeight": "28rem",
}
`;

exports[`Min Height 6`] = `
Object {
  "minHeight": "100vh",
}
`;

exports[`Min Height 7`] = `
Object {
  "minHeight": "1024px",
}
`;

exports[`Min Height 8`] = `
Object {
  "minHeight": "min-content",
}
`;

exports[`Min Height 9`] = `
Object {
  "minHeight": "0.875rem",
}
`;

exports[`Min Height 10`] = `
Object {
  "minHeight": "100%",
}
`;

exports[`Min Width 1`] = `
Object {
  "width": "min-content",
}
`;

exports[`Min Width 2`] = `
Object {
  "minWidth": "20%",
}
`;

exports[`Min Width 3`] = `
Object {
  "minWidth": "25%",
}
`;

exports[`Min Width 4`] = `
Object {
  "minWidth": "2.5rem",
}
`;

exports[`Min Width 5`] = `
Object {
  "minWidth": "28rem",
}
`;

exports[`Min Width 6`] = `
Object {
  "minWidth": "100vw",
}
`;

exports[`Min Width 7`] = `
Object {
  "minWidth": "1024px",
}
`;

exports[`Min Width 8`] = `
Object {
  "minWidth": "min-content",
}
`;

exports[`Min Width 9`] = `
Object {
  "minWidth": "0.875rem",
}
`;

exports[`Min Width 10`] = `
Object {
  "minWidth": "100%",
}
`;

exports[`Width 1`] = `
Object {
  "width": "20%",
}
`;

exports[`Width 2`] = `
Object {
  "width": "25%",
}
`;

exports[`Width 3`] = `
Object {
  "width": "2.5rem",
}
`;

exports[`Width 4`] = `
Object {
  "width": "28rem",
}
`;

exports[`Width 5`] = `
Object {
  "width": "100vw",
}
`;

exports[`Width 6`] = `
Object {
  "width": "1024px",
}
`;

exports[`Width 7`] = `
Object {
  "width": "min-content",
}
`;

exports[`Width 8`] = `
Object {
  "width": "0.875rem",
}
`;

exports[`Width 9`] = `
Object {
  "width": "100%",
}
`;
